src/pages/index.tsx

```diff
@@ -39,6 +36,25 @@ const Dashboard: NextPage = () => {
   const [pluginIds, setPluginIds] = useState<PluginIdDocument[]>([]);
   const [selectedPluginId, setSelectedPluginId] = useState<string>("");
   const [selectedYear, setSelectedYear] = useState<string>("2022");
+  const [hasMonetizationListener, setMonetizationListener] = useState<boolean>(false);
+  const batcher = new Batcher();
+
+  useEffect(() => {
+    if (!hasMonetizationListener) {
+      setMonetizationListener(true);
```

Comment:
If you are already setting hasMonetizationListener to true whener the app renders, why do you need the state hasMonetizationListener?

Reply:
Actually I set to false as default, and I did this validation to not overwrite the listener.

Reply:
Ok. You set the default to false. But the useEffect runs as soon as the page is loaded and set it to true. Why not set the default value to true?

Reply:
Move batcher to inside the useEffect